packager/packager.cc

```diff
@@ -1615,18 +1615,33 @@ unique_ptr<PackageInfoImpl> definePackage(const core::GlobalState &gs, ast::Pars
         // Other than being able to say "we don't mutate the trees in packager" there's not much
         // value in going that far (even namer mutates the trees; the packager fills a similar role).
 
-        // Pre-resolve the super class. This makes it easier to detect that this is a package
-        // spec-related class def in later passes without having to recursively walk up the constant
-        // lit's scope to find if it starts with <PackageSpecRegistry>.
-        superClass = ast::make_expression<ast::ConstantLit>(core::Symbols::PackageSpec(),
-                                                            superClass.toUnique<ast::UnresolvedConstantLit>());
-
         auto nameTree = ast::cast_tree<ast::UnresolvedConstantLit>(packageSpecClass->name);
         if (!validatePackageName(ctx, nameTree)) {
             reportedError = true;
+
+            // "Remove" the superclass.
+            //
+            // `::PackageSpec` doesn't exist. Normally we would rewrite it
+            // to Sorbet::Private::Static::PackageSpec, but we don't do that if the package spec
+            // definition is invalid.
+            //
+            // To avoid the chance that the user only sees the "Unable to resolve PackageSpec" error
+            // and then gets confused, we "remove" the super class here, treating it like the user
+            // omitted the superclass.
+            //
+            // This establishes an invariant that if the superClass is a resolved constant and
+            // equal to Symbols::PackageSpec(), then it's the canonical package def in this file
+            superClass = ast::MK::Constant(packageSpecClass->loc, core::Symbols::todo());
```

Comment:
Why a `todo` instead of an `EmptyTree`?

Reply:
This mimics how we desugar ClassDef's. If we did EmptyTree, it would not eventually descend from `Object` like other ClassDef's that don't have an explicit ancestor.